core/src/test/java/fj/IOTest.java

```diff
@@ -0,0 +1,34 @@
+package fj;
+
+import fj.data.IO;
+import fj.data.IOFunctions;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+
+public class IOTest {
+
+    String result = null;
+
+    @Test
+    public void testLift() throws IOException {
+        final IO<String> readName = () -> new BufferedReader(new StringReader("foo")).readLine();
+        final F<String, IO<Unit>> upperCaseAndPrint = F1Functions.<String, IO<Unit>, String>o(this::println).f(String::toUpperCase);
+        final IO<Unit> readAndPrintUpperCasedName = IOFunctions.bind(readName, upperCaseAndPrint);
+        readAndPrintUpperCasedName.run();
+        assertThat(result, is("FOO"));
+    }
+
+    public IO<Unit> println(final String s) {
```

Comment:
Would it make sense to make this return the string, rather than assign it to the instance variable `result`.

Longer term, perhaps we need a bind method like: `IO<A> bindLeft(IO<A> io, F<A, IO<B>> f)` which binds, but returns the a from the left argument.  But this would be another issue.  I would have to check the Haskell doc to see if this exists.